### PR TITLE
Run installer with enable katello-agent flags for Katello 4.1+

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: bats/fb-katello-client.bats [hunk 1]
```diff
@@ -78,6 +78,21 @@ load fixtures/content
...
+@test "try fetching container content" {
+  tPackageExists podman || tPackageInstall podman
+  podman login $HOSTNAME -u admin -p changeme
+  CONTAINER_PULL_LABEL=`echo "${ORGANIZATION_LABEL}-${PRODUCT_LABEL}-${CONTAINER_REPOSITORY_LABEL}"| tr '[:upper:]' '[:lower:]'`
+  podman pull "${HOSTNAME}/${CONTAINER_PULL_LABEL}"
+}
+
+@test "enable katello-agent" {
+  KATELLO_VERSION=$(tKatelloVersion)
+  if [[ $KATELLO_VERSION >= 4.1 ]]; then
```
Comment (evgeni) [inline bats/fb-katello-client.bats:90]: I don't think that's a valid test expression.

File: bats/fb-katello-client.bats [hunk 2]
```diff
@@ -78,6 +78,21 @@ load fixtures/content
   done
 }
 
+@test "try fetching container content" {
+  tPackageExists podman || tPackageInstall podman
+  podman login $HOSTNAME -u admin -p changeme
+  CONTAINER_PULL_LABEL=`echo "${ORGANIZATION_LABEL}-${PRODUCT_LABEL}-${CONTAINER_REPOSITORY_LABEL}"| tr '[:upper:]' '[:lower:]'`
```
Comment (evgeni) [inline bats/fb-katello-client.bats:84]: unrelated to this PR, but it feels like this should live in https://github.com/theforeman/forklift/blob/master/bats/fixtures/content.bash?

**Review by evgeni — APPROVED** [review]

File: bats/fb-katello-client.bats [hunk 3]
```diff
@@ -78,6 +78,21 @@ load fixtures/content
...
+  podman pull "${HOSTNAME}/${CONTAINER_PULL_LABEL}"
+}
+
+@test "enable katello-agent" {
+  KATELLO_VERSION=$(tKatelloVersion)
+  if [[ $KATELLO_VERSION != 4.[1-9]* ]]; then
+    skip "Enabling katello-agent explicitly is only available with Katello 4.1+"
+  fi
+  foreman-installer --katello-enable-katello-agent true --foreman-proxy-content-enable-katello-agent true
```
Comment (ekohl) [inline bats/fb-katello-client.bats:93]: Just a thought about how we expose it (https://github.com/theforeman/puppet-katello/pull/387): we can not expose it in init.pp but rather in `katello::params`. Then in the Hiera config we set:
```yaml
---
katello::params::enable_katello_agent: "%{lookup('foreman_proxy_content::enable_katello_agent')}"
```

The result would be that there's a single parameter that works both for all scenarios: `--foreman-proxy-content-enable-katello-agent true`. Now that's my theory and I haven't tested it, but it's a path I'd like to investigate.

Comment (evgeni) [reply bats/fb-katello-client.bats:93]: Semi related: can we do the same for all plugins? :)

Comment (ehelms) [reply bats/fb-katello-client.bats:93]: I can certainly give it a try.

Comment (ehelms) [reply bats/fb-katello-client.bats:93]: I gave this a try and am running into the following:

```
Error: Evaluation Error: Error while evaluating a Function Call, Class[Katello::Params]: parameter 'enable_katello_agent' expects a Boolean value, got String (line: 120, column: 346) on node pipe-katello-server-nightly-centos7.wareagle.example.com
```

With this defined:

```
katello::params::enable_katello_agent: lookup('foreman_proxy_content::enable_katello_agent')
```

Comment (ehelms) [reply bats/fb-katello-client.bats:93]: I read that this issue might be related to how yaml format for Hiera handle boolean types (or lack thereof).

Comment (ekohl) [reply bats/fb-katello-client.bats:93]: https://puppet.com/docs/puppet/5.5/hiera_automatic.html#arguments suggests a second argument can be passed to lookup. That determines the data type. Haven't tried it myself.

**Pull request merged**